lib/ansible/plugins/connection/local.py

```diff
@@ -80,7 +80,7 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
         p = subprocess.Popen(
             cmd,
-            shell=isinstance(cmd, basestring),
+            shell=isinstance(cmd, text_type),
```

Comment:
This one is wrong... needs to account for both text strings and byte strings.  So probably have to compose it from individual constants in six.

``` python
shell=isinstance(cmd,(text_type, binary_type))
```